subsys/net/ip/tcp.c

```diff
@@ -392,6 +396,47 @@ static void tcp_derive_rto(struct tcp *conn)
 #endif
 }
 
+#ifdef CONFIG_NET_TCP_CONGESTION_AVOIDANCE
+static void tcp_new_reno_init(struct tcp *conn)
+{
+	conn->ca.congestion_win = conn_mss(conn) * TCP_CONGESTION_INITIAL_WIN;
+	conn->ca.ssthresh = conn_mss(conn) * TCP_CONGESTION_INITIAL_SSTHRESH;
+	conn->ca.state = TCP_NEW_RENO_RAMPUP;
+}
+
+static void tcp_new_reno_fast_retransmit(struct tcp *conn)
+{
+	conn->ca.congestion_win = MAX(conn_mss(conn), conn->ca.congestion_win / 2);
```

Comment:
Could you reference the RFC/chapter where the formulas come from? It should simplify future maintenance.

Reply:
Done